README.md

```diff
@@ -1,2 +1,307 @@
+
+2. `--newick`: Newick-format tree, in which labels of nodes (tips, internal nodes and root) are considered as taxa. All nodes must have labels and all labels must be unique.
+
+3. `--ranktb`: Table of per-taxon per-rank assignments. Each column represents a rank. Column header will be treated as rank name.
+
+4. `--lineage`: Map of taxon to lineage string (`;`-delimited taxa from high to low)
+
+   Can be **Greengenes**-style taxonomy where rank codes such as `k__` will be parsed. But the rank code is not mandatory. Unassigned taxon (e.g., `s__`) and non-unique taxon are acceptable (e.g., `p__Actinobacteria` and `c__Actinobacteria`).
+
+   Compatible with widely-used taxonomy systems in e.g., QIIME, SHOGUN, MetaPhlAn2, GTDB, etc.
+
+5. `--map` or `-m`: Simple map of lower taxon to higher taxon. Filename stem represents rank name.
+
+   Can supply multiple maps (by typing multiple `--map` parameters) to constitute several hierarchies. For example, the 1st file maps genes to UniRef entries, the 2nd maps UniRef entries to GO terms, the 3rd maps GO terms to GO slim terms, so on so forth.
+
+If no classification file is provided, Woltka will automatically build a classification system from the alignment files, in which subject identifiers will be parsed as lineage strings.
+
+Subjects themselves are part of the classification system. A map of subject to taxon (e.g., a genome ID to NCBI TaxID map) can be supplied with the `--map` parameter if necessary.
+
+Furthermore, one can supply Woltka with a taxon name dictionary, and the output profile will show taxon names instead of taxon IDs:
+
+* `--names`: NCBI-style `names.dmp` or a simple taxon-to-name map.
+
+Classification files are **additive** --- unless they conflict --- which will be noted by Woltka. Example command:
+
+```bash
+woltka classify \
+  --map nucl2g.txt \
+  --map g2taxid.txt \
+  --names taxdump/names.dmp \
+  --nodes taxdump/nodes.dmp \
+  ...
+```
+
+Again, compressed files are supported and automatically recognized. The following command works:
+
+```bash
+woltka classify \
+  --lineage gg_13_5_taxonomy.txt.gz
+  ...
+```
+
+
+## Combined taxonomic & functional analyses
+
+Woltka combines the two fundamental analyses in metagenomics: taxonomic profiling (mapping reads to genomes) and functional profiling (mapping reads to functional genes) into one run. This saves compute, ensures consistency, and allows for stratification which is essential for understanding functional diversity across the microbial community.
+
+This is achieve by an efficient algorithm implemented in Woltka, which matches read alignments and annotated genes based on their coordinates on the host genome.
+
+The coordinates of read-to-genome alignments are provided in the alignment files. One needs to provide Woltka with a table of gene coordinates. The format is like:
+
+WoL reannotations (available for [download](https://biocore.github.io/wol/)):
+
+```
+>G000006745
+1       806     372
```

Comment:
It is unclear to me what these coordinates actually mean, or where one would get such annotations.

Reply:
Response to the general comments below. Restructured documentation in commit [f7f2b34](https://github.com/qiyunzhu/woltka/pull/1/commits/f7f2b34e295140b78406fdbe366412afde0c1d10). Table of contents expanded. Sections moved to separate files. Example usage expanded to include overview of input / output files.

> ## README.md
> 
> looks good. comments inline.
> 
> some general feedback: I think careful attention should be paid to ordering. Specifically, I think it may be nice to have an overview of expected outputs (so users know why they would want to use your tool) and expected inputs (so users know what they need in order to use your tool) at the beginning of the readme, maybe starting right after your table of contents. This can brief compared to the more detailed descriptions you give below.